src/pages/Welcome/Welcome.jsx

```diff
@@ -0,0 +1,38 @@
+import { faAngleRight } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useRef, useState } from "react";
+import "./welcome.css";
+
+export default function Welcome() {
+  const nameInput = useRef("");
+  const [name, setName] = useState("");
+
+  return (
+    <div className="welcome-container background-image">
+      <div className="welcome-text">Hello! What's your name?</div>
+      <input
+        type="text"
+        className="welcome-input"
+        ref={nameInput}
+        onChange={() => {
+          setName(nameInput.current.value);
+        }}
+      />
+
+      {name !== "" ? (
+        <button
+          className="welcome-button"
+          onClick={() => {
+            localStorage.setItem("userName", nameInput.current.value);
+            window.location.reload(false);
```

Comment:
can you explain what this is doing ?

Reply:
Basically, for the extension to be hosted on a web store it cannot have routes
So here it's setting the user name in local storage and triggering a reload to read that state and render the appropriate page ( the main page if the name is already set or the welcome page if the user still hasn't been set)

Reply:
okay, nice.